modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointWorkflow.java

```diff
@@ -289,31 +282,26 @@ private DataRegionsDirtyPages beginCheckpoint(
             Collection<? extends DataRegion<PersistentPageMemory>> dataRegions,
             CompletableFuture<?> allowToReplace
     ) throws IgniteInternalCheckedException {
-        Collection<DataRegionDirtyPages<CollectionDirtyPages>> dataRegionsDirtyPages = new ArrayList<>(dataRegions.size());
+        Collection<DataRegionDirtyPages<DirtyPagesCollection>> dataRegionsDirtyPages = new ArrayList<>(dataRegions.size());
```

Comment:
Can I ask you to add assertions that write lock is locked in every method that needs it? Currently it's impossible to understand it by just looking at method names or javadocs

Reply:
I'll try to do it.